docs/design/key_classes.md

```diff
@@ -0,0 +1,55 @@
+# Key classes
+
+## `DetectedObject`
+
+This class represents a detected object received from the host agent (HA) or a remote agent (RA).
+
+### Member variables
+
+* `timestamp`: global time associated with the `state`
+* `state`: state vector (CTRV model)
+* `state_covariance`: covariance matrix associated with `state`
+
```

Comment:
Is there any additional meta data which needs to be tracked with the object beyond these core fields?

Reply:
I can immediately think of a reference frame field that should be tracked eventually. However, since we're only working on host object tracking, we can assume the `state` is in the host's body frame. Once we start adding remote object tracking, we can modify the `DetectedObject` class accordingly.